article/_posts/2021-09-02-sindyautoencoder.md

```diff
@@ -0,0 +1,154 @@
+autoencoders to discover the coordinate system and SINDy to find the sparse dynamical model in the autoencoder's 
+latent space.
+
+If the data $$x(t)$$ is high dimensional, the goal is to find the function $$\mathcal{g}$$: 
+
+$$
+\frac{d}{dt} z(t) = \mathbb{g}(z(t))
+$$
+
+where $$z(t) = \varphi(x(t)) \in \mathcal{R}^d (d \ll n)$$
+
+
+SINDy algorithm is applied on the latent variables to find the governing equations in the latent space with appropriate 
+basis functions.
+
+$$
+\frac{d}{dt} z(t) = \mathbb{g}(x(t)) = \Theta(z(t))\Xi
+$$
+
+The method is trained with 4 loss terms. 
+
+* Reconstruction loss
+* SINDy loss in $$\dot{x}$$
+* SINDy loss in $$\dot{z}$$
+* SINDy regularization
+
+![](/article/images/sindyautoencoder/fig1.jpg)
+
+
+The latent variable time derivatives can be computed by $$\dot{z}(t) = \nabla_x\varphi(x(t))\dot{x}(t)$$. (Essentially 
+forward propagation of the derivates of $$x$$). 
+
+To achieve a truly sparse model, coefficients in $$\Xi$$ under $$0.1$$ are set to zero for the remainder of training at 
+every 500 epochs. 
+
+
+# Results
+
+
+Authors test their method on 3 systems with well know equations. They generate synthetic high dimensional data and
+train their autoencoder to find the underlying coordinate system and equations.   
+
+![](/article/images/sindyautoencoder/fig2.jpg)
+
+## Chaotic Lorenz system
+
+The authors construct a high dimensional example base on the chaotic Lorenz system defined by: 
+
+![](/article/images/sindyautoencoder/lorenzeq8.jpg)
+
+The high dimensional data is created with 6 fixes spatial modes $$u_1,..., u_6 \in \mathcal{R}^{128}$$ given by 
+Legendre polynomial: 
+
+![](/article/images/sindyautoencoder/lorenzeq9.jpg)
+
+Their model discovers a correct model and given a affine transformation the authors recover the original model. 
+
+![](/article/images/sindyautoencoder/fig3a.jpg)
+
+## Reaction-diffusion
+
+Authors test their method on the lambda-omega reaction-di~~~~ffusion system and obtain the following results:
+
+![](/article/images/sindyautoencoder/figs2.jpg)
+
+
+## Nonlinear pendulum
+
+The non-linear pendulum is governed by a second order equation: 
+
+$$
+\ddot{z} = - \sin z
+$$
+
+Authors generate snapshot images with a two dimensional Gaussian centered at the center of mass of the pendulum given 
+by the angle z.  
+
+In theory, a linear autoencoder (PCA) could represent this data, but in practice a non-linear autoencoder is required. The 
+autoencoder has a latent dimension of 1. The second order derivatives must be computed for $$x$$ and propagated to the latent space. 
+
+They obtain the following models: 
+
+![](/article/images/sindyautoencoder/figs3.jpg)
+
+
+# Conclusions
+Authors point out the this method requires approximately noise-free data because their method requires good estimates 
+of the derivatives.
+
+Code is available here: https://github.com/kpchamp/SindyAutoencoders
```

Comment:
Maybe make the URL into a link so it's easily clickable?
```suggestion
Code is available here: [https://github.com/kpchamp/SindyAutoencoders](https://github.com/kpchamp/SindyAutoencoders)
```

Reply:
Fixed in cdcfddf